### DISCUSSION TITLE
Error outputting keys and certificates

### DISCUSSION THREAD

**Discussion by life777eternal** [openssl/openssl#23547] [Q&A] [answered]
Installed OpenSSL with Choco and after I tried the commands specified [here](https://encryptionconsulting.com/how-to-convert-pfx-encoded-certificate-file-to-pem-format-using-openssl/) it gave this error. 
```
C:\Users\*******>openssl pkcs12 -in C:/*******.pfx -clcerts -nokeys -out C:/*******crt.pem
Enter Import Password:
Error outputting keys and certificates
90400000:error:0308010C:digital envelope routines:inner_evp_generic_fetch:unsupported:crypto\evp\evp_fetch.c:355:Global default library context, Algorithm (RC2-40-CBC : 0), Properties ()
```

Greetings, I found out from a [WinAmp Forum post](https://forums.winamp.com/forum/shoutcast/shoutcast-technical-support/315804-perfect-ssl-shoutcast-server) that the latest version of the SHOUTcast server can use an SSL/TLS Connection, but with pem files and not pfx files. Although the WinAcme (2.2.6) that I have on my Windows server didn't save the new cert as a pem file.  It saved a ...-csr.pem and a ...-temp.pfx file. 

Also I have some OpenSim regions on my server connected to Osgrid, and I have the SSL/TLS connection setup for them, and I've had to [import the private key](https://alitajran.com/export-lets-encrypt-certificate-in-windows-server/) into the windows store, and export the pfx cert, but a pfx file apparently won't work on the [SHOUTcast](https://download.nullsoft.com/shoutcast/tools/sc_serv2_win64-latest.exe) 'sc_serve.exe' program. 

Thank you, Shalom.

**Comment by t8m** [comment, marked as answer]
Please try to use the `-legacy` option of the `pkcs12` app.

**Reply by life777eternal** [reply]
Oh alright, so you want me to put in "-legacy" after the "pkcs12" and before the "-in"?
Thank you.

**Reply by lhztop** [reply]
yes

**Comment by robkorv** [comment]
For commands that do not support the "-legacy" option, you can enable legacy providers in your ssl config. I recommend to disable it again after the legacy file is decrypted and before you re-encrypt the file. You can check which providers are active with `openssl list -providers`

```diff
--- /etc/ssl/openssl.cnf.bak
+++ /etc/ssl/openssl.cnf
@@ -56,6 +56,7 @@
 # List of providers to load
 [provider_sect]
 default = default_sect
+legacy = legacy_sect
 # The fips section name should match the section name inside the
 # included fipsmodule.cnf.
 # fips = fips_sect
@@ -69,9 +70,10 @@
 # OpenSSL may not work correctly which could lead to significant system
 # problems including inability to remotely access the system.
 [default_sect]
-# activate = 1
-
+activate = 1
 
+[legacy_sect]
+activate = 1
 ####################################################################
 [ ca ]
 default_ca     = CA_default            # The default ca section
```

I used https://www.practicalnetworking.net/practical-tls/openssl-3-and-legacy-providers/ as a reference.
